### ISSUE TITLE
AUC-PR calculation for emty masks?

### ISSUE THREAD

**Issue by osivaz61** [CaglarGuher/IDRiD-Eye-Fundus-Dataset-Lesion-Segmentation#2]
Hi, 

Some images do not have lesions (some SE lesions images have empty masks).  How did you handle this when calculating the AUC-PR for empty masks?

Also, can you share the dice score results?

Thanks

**Comment by CaglarGuher** [comment]
The algorithm takes every test images with corresponding groundtruths and combine into one big image then calculates the auprc , as the previous papers suggested.

**Comment by osivaz61** [comment]
I want to ask to be sure. Let's assume we have 24 images whose sizes are 400x400. Do you want to say aucpr get parameters like this aucpr(pred, gt) where pred and gt are 3840000x1 arrays?

3 840 000 = 400x400x24

**Comment by CaglarGuher** [comment]
More like if you have 24 images with 400*400 resolution , the final image
with corresponding gt will be 96000*96000 (((400*24)*(400*24))) ,, because
the  final image is too big , you will need min of 16gb ram to run this
aucpr calculation without any problem.

**Comment by osivaz61** [comment]
Actually, I am trying to understand the scenario that does not use a patch mechanism. Normally without a patch approach, there are 54 train images and 27 test images in the IDRID dataset. However, the number of images including SE lesions is 27 and 14 for the train and test sets respectively. 

When we want to segment all lesions with one model (multi-class) we have to use all train and test images. The point I am trying to understand is how can I handle the empty mask scenario when I want to calculate aucpr for SE lesion.

Because the some of 27 test images include SE lesion, some of not.

**Comment by CaglarGuher** [comment]
You basicly create black empty groundtruths for the model  (for example
with paint) because no groundtruths means there is no lesion for that
spesific image.I hope I understood your question:)

12 Şub 2024 Pzt 19:29 tarihinde osivaz61 ***@***.***> şunu
yazdı:

**Comment by osivaz61** [comment]
When I want to calculate aucpr for an empty mask and predicted values, the method that calculates aucpr gives a warning like that 'the mask should contain 1s' because all values are 0. I want to ask how you handle this problem when you apply multi-class segmentation.

It is reasonable to set aucpr to 1 or 0 for an empty mask? Because I am trying to apply multi-class segmentation?

Thanks

**Comment by osivaz61** [comment]
Thanks to you I investigated the original PBDA github code and I found that the gitHub code of the PBDA paper calculates the auc-pr by thinking of all test images (27) as one image. Not calculating aucpr for every image.

There is a different pixel-level retina dataset whose name is FGADR. I am trying to apply multi-class segmentation to it. But there are also empty masks in the FGADR dataset. I can not reproduce the result of FGADR dataset paper. Did you try to segment the FGADR dataset before?
